stories/core/Breadcrumbs.stories.tsx

```diff
@@ -0,0 +1,164 @@
+  title: 'Core/Breadcrumbs',
+  argTypes: {
+    className: { control: { disable: true } },
+    style: { control: { disable: true } },
+    id: { control: { disable: true } },
+    children: { control: { disable: true } },
+  },
+} as Meta<BreadcrumbsProps>;
+
+export const Basic: Story<BreadcrumbsProps> = (args) => {
+  return (
+    <Breadcrumbs {...args}>
+      <BreadcrumbItem key={0} onClick={() => action('Root')()}>
+        Root
+      </BreadcrumbItem>
+      <BreadcrumbItem key={1} onClick={() => action('Item 1')()}>
+        Item 1
+      </BreadcrumbItem>
+      <BreadcrumbItem key={2} onClick={() => action('Item 2')()}>
+        Item 2
+      </BreadcrumbItem>
+    </Breadcrumbs>
+  );
+};
+
+export const Links: Story<BreadcrumbsProps> = (args) => {
+  return (
+    <Breadcrumbs {...args}>
+      <BreadcrumbItem key={0} element='a' href='/'>
+        iTwinUI
+      </BreadcrumbItem>
+      <BreadcrumbItem key={1} element='a' href='/?path=/docs/core-breadcrumbs'>
+        Breadcrumbs
+      </BreadcrumbItem>
+      <BreadcrumbItem key={2} element='span'>
+        Links
+      </BreadcrumbItem>
+    </Breadcrumbs>
+  );
+};
+
+export const CustomSeparator: Story<BreadcrumbsProps> = (args) => {
+  return (
+    <Breadcrumbs separator={<SvgChevronRightDouble />} {...args}>
+      <BreadcrumbItem key={0} onClick={() => action('Root')()}>
+        Root
+      </BreadcrumbItem>
+      <BreadcrumbItem key={1} onClick={() => action('Item 1')()}>
+        Item 1
+      </BreadcrumbItem>
+      <BreadcrumbItem key={2} onClick={() => action('Item 2')()}>
+        Item 2
+      </BreadcrumbItem>
+    </Breadcrumbs>
+  );
+};
+CustomSeparator.args = {
+  separator: <SvgChevronRightDouble />,
+};
+
+export const Overflow: Story<BreadcrumbsProps> = (args) => {
+  const items = Array(10)
+    .fill(null)
+    .map((_, index) => (
+      <BreadcrumbItem key={index}>Item {index}</BreadcrumbItem>
+    ));
+
+  return (
+    <div style={{ maxWidth: '50%', border: '1px solid lightpink', padding: 8 }}>
+      <Breadcrumbs {...args}>{items}</Breadcrumbs>
+    </div>
+  );
+};
+
+export const FolderNavigation: Story<BreadcrumbsProps> = (args) => {
+  const items = React.useMemo(
+    () => ['Root', 'Level 1', 'Level 2', 'Level 3', 'Level 4'],
+    [],
+  );
+
+  const [lastIndex, setLastIndex] = useState(items.length - 1);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const breadcrumbItems = React.useMemo(
+    () =>
+      items.slice(0, lastIndex + 1).map((item, index) => (
+        <BreadcrumbItem
+          key={`Breadcrumb${index}`}
+          onClick={() => {
+            if (lastIndex !== index) {
+              setLastIndex(index);
+            } else {
+              setIsEditing(true);
+            }
+          }}
+        >
+          {item}
+        </BreadcrumbItem>
+      )),
+    [items, lastIndex],
+  );
+
+  return (
+    <div style={{ display: 'inline-flex', width: 450 }}>
+      {isEditing ? (
+        <Input
```

Comment:
I thought we will handle editing as it is in standards. Ofc it is not needed in initial implementation so I don't think this should appear in a story.

Reply:
But you asked me to add a story for this folder navigation pattern...

We cannot handle editing in Breadcrumbs because it is literally a different component. This story is just showing how the user might be able to do it.